<!DOCTYPE html><html><head>
  <title>TC39 Async Context on 2024-05-28</title>
  <link rel="stylesheet" href="../style.css">
  <script>window.room = "TC39 Async Context";</script><script src="../render-roomlist.js"></script>
  <script src="../logs.js"></script>
</head>
<body><div class="wrapper">
<div class="sidebar">
<div class="title">TC39 Async Context<br>2024-05-28<br><a href="plaintext/">plaintext logs</a></div>
<div class="nav">
<a href="2024-05-27" class="nav-link"><span>prev</span></a>
<span style="float:right">next</span>
</div>
    <div class="all-rooms"><noscript>JavaScript is required to load the channel index, but you can go to <a href="../..">the static index</a> directly.</noscript></div>
<div class="footer"><a href="https://github.com/bakkot/matrix-archive-bot">source on github</a></div>
</div>
<div class="rhs">
<div class="rhs-header">
<span id="error" style="color: red; display:none">error</span>

<input type="text" id="query" size="25" placeholder="Search TC39 Async Context">
<a id="search-submit" class="button icon-link" title="Search">
  <svg class="icon" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><path d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"></path></svg>
</a>
</div>

<div class="log"><table id="log-table"><tbody id="log-tbody">

  <tr class="msg" id="L0"><td class="ts-cell"><a class="ts" href="#L0" alt="Tue May 28 2024 02:46:20 GMT-0700 (Pacific Daylight Time)">09:46</a></td><td class="nick-cell"><div class="m-ov">&lt;<span class="nick nick-4" title="@abotella:igalia.com">Andreu Botella</span>&gt;</div></td><td class="msg-cell"><mx-reply><blockquote> A single promise could be branched multiple times, and <code>unhandledrejection</code> events are dispatched for each "unhandled" promise, rather than a single source of rejection</blockquote></mx-reply>I don't think this is a problem. Context splits are not problematic, only merges are. If you only care about one of the branched promises but not the rest, you could still want to know the source of rejection</td></tr>
  <tr class="msg" id="L1"><td class="ts-cell"><a class="ts" href="#L1" alt="Tue May 28 2024 02:47:41 GMT-0700 (Pacific Daylight Time)">09:47</a></td><td class="nick-cell"><div class="m-ov">&lt;<span class="nick nick-4" title="@abotella:igalia.com">Andreu Botella</span>&gt;</div></td><td class="msg-cell">the way I think of it, if an exception (or promise rejection) is "automatically rethrown" (as might happen conceptually in <code>run</code>), then if you have a way to act on that error, then that context shouldn't be lost without a trace</td></tr>
  <tr class="msg" id="L2"><td class="ts-cell"><a class="ts" href="#L2" alt="Tue May 28 2024 02:48:20 GMT-0700 (Pacific Daylight Time)">09:48</a></td><td class="nick-cell"><div class="m-ov">&lt;<span class="nick nick-4" title="@abotella:igalia.com">Andreu Botella</span>&gt;</div></td><td class="msg-cell">if you choose to catch an exception/rejection and rethrow it, then you might choose to use the original throw context, or switch it depending on your use case</td></tr>
  <tr class="msg" id="L3"><td class="ts-cell"><a class="ts" href="#L3" alt="Tue May 28 2024 02:48:43 GMT-0700 (Pacific Daylight Time)">09:48</a></td><td class="nick-cell"><div class="m-ov">&lt;<span class="nick nick-4" title="@abotella:igalia.com">Andreu Botella</span>&gt;</div></td><td class="msg-cell">the same way you'd wrap with a higher-level exception at API boundaries</td></tr>
  <tr class="msg" id="L4"><td class="ts-cell"><a class="ts" href="#L4" alt="Tue May 28 2024 02:48:58 GMT-0700 (Pacific Daylight Time)">09:48</a></td><td class="nick-cell"><div class="m-ov">&lt;<span class="nick nick-4" title="@abotella:igalia.com">Andreu Botella</span>&gt;</div></td><td class="msg-cell">the inner throw context would be an implementation detail</td></tr>
  <tr class="msg" id="L5"><td class="ts-cell"><a class="ts" href="#L5" alt="Tue May 28 2024 02:49:06 GMT-0700 (Pacific Daylight Time)">09:49</a></td><td class="nick-cell"><div class="m-ov">&lt;<span class="nick nick-4" title="@abotella:igalia.com">Andreu Botella</span>&gt;</div></td><td class="msg-cell">am I making sense?</td></tr>
  <tr class="msg" id="L6"><td class="ts-cell"><a class="ts" href="#L6" alt="Tue May 28 2024 02:53:43 GMT-0700 (Pacific Daylight Time)">09:53</a></td><td class="nick-cell"><div class="m-ov">&lt;<span class="nick nick-7" title="@legendecas:matrix.org">Chengzhong Wu</span>&gt;</div></td><td class="msg-cell"><mx-reply><blockquote> the way I think of it, if an exception (or promise rejection) is "automatically rethrown" (as might happen conceptually in <code>run</code>), then if you have a way to act on that error, then that context shouldn't be lost without a trace</blockquote></mx-reply>What's an "automatical rethrown"?</td></tr>
  <tr class="msg" id="L7"><td class="ts-cell"><a class="ts" href="#L7" alt="Tue May 28 2024 02:55:03 GMT-0700 (Pacific Daylight Time)">09:55</a></td><td class="nick-cell"><div class="m-ov">&lt;<span class="nick nick-4" title="@abotella:igalia.com">Andreu Botella</span>&gt;</div></td><td class="msg-cell">well, I was thinking of how <code>run()</code> essentially rethrows a thrown exception, or how a <code>.then()</code> without a catch handler essentially rethrows a promise rejection, without any user code</td></tr>

</tbody></table></div></div></div>
</body></html>